### PR TITLE
feat: Added referrer for login

### LINKED ISSUE #138: Redirect to requested URL after login
## Description
If a non-logged in user attempts to load a page (i.e. https://app.staging.terraso.org/landscapes), they are redirected to the login page.

## Actual behavior
After login, user is show the home page.

## Expected behavior
After login, user is redirected to desired page, ie. https://app.staging.terraso.org/landscapes.

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Added referrer for login process

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [ ] Strings are localized
- [x] Verified on mobile
- [x] Verified on desktop

### Related Issues
Fixes #138 

### Verification steps

Screenshot:

referrer

### CODE REVIEW

File: src/account/components/AccountLogin.js [hunk 1]
```diff
@@ -10,10 +11,15 @@ import PageLoader from 'common/components/PageLoader';
 
 import logo from 'assets/logo.svg';
 
+const appendReferrer = (url, referrer) =>
+  referrer ? `${url}&state=${referrer}` : url;
```
Comment (paulschreiber) [inline src/account/components/AccountLogin.js:15]: issue: use a more descriptive variable name than `state`

File: src/account/components/AccountLogin.js [hunk 2]
```diff
@@ -10,10 +11,15 @@ import PageLoader from 'common/components/PageLoader';
...
+
 const AccountForm = () => {
   const dispatch = useDispatch();
   const { t } = useTranslation();
+  const [searchParams] = useSearchParams();
   const { fetching, urls } = useSelector(state => state.account.login);
+  const referrer = searchParams.get('referrer');
```
Comment (paulschreiber) [inline src/account/components/AccountLogin.js:22]: question: should this query parameter match L15 above?

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/account/components/AccountLogin.js (hunk 1 above)
Comment (josebui) [reply src/account/components/AccountLogin.js:15]: @paulschreiber This was decided on the backend side, for context:
- Bakend PR: https://github.com/techmatters/terraso-backend/pull/116
- Slack conversation: https://tech-matters.slack.com/archives/C02K02L3Q48/p1643385309308109
Tagging @cacarrara for vis

File: src/account/components/AccountLogin.js (hunk 2 above)
Comment (josebui) [reply src/account/components/AccountLogin.js:22]: @paulschreiber This is the query parameter exposed in the browser, the one in L15 is the one sent to the API

**Comment by paulschreiber** [comment]
@cacarrara can you look this over?

File: src/account/components/AccountLogin.js (hunk 1 above)
Comment (cacarrara) [reply src/account/components/AccountLogin.js:15]: @paulschreiber I'm ok with `state`. This is the default variable name for this purpose.

File: src/account/components/AccountLogin.js (hunk 2 above)
Comment (cacarrara) [reply src/account/components/AccountLogin.js:22]: I have no preference using `referrer` or something else here. So, `referrer` is ok.

**Review by cacarrara — APPROVED** [review]
I'm ok with this implementation. :+1:

**Review by paulschreiber — APPROVED** [review]

**Review by cacarrara — APPROVED** [review]

**Pull request merged**